x-pack/plugin/migrate/src/main/java/org/elasticsearch/xpack/migrate/action/ReindexDataStreamIndexAction.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+package org.elasticsearch.xpack.migrate.action;
+
+import org.elasticsearch.action.ActionRequest;
+import org.elasticsearch.action.ActionRequestValidationException;
+import org.elasticsearch.action.ActionResponse;
+import org.elasticsearch.action.ActionType;
+import org.elasticsearch.action.IndicesRequest;
+import org.elasticsearch.action.support.IndicesOptions;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+
+import java.io.IOException;
+import java.util.Objects;
+
+public class ReindexDataStreamIndexAction extends ActionType<ReindexDataStreamIndexAction.Response> {
+
+    public static final String NAME = "indices:admin/migration/index/reindex";
+
+    public static final ActionType<Response> INSTANCE = new ReindexDataStreamIndexAction();
+
+    private ReindexDataStreamIndexAction() {
+        super(NAME);
+    }
+
+    public static class Request extends ActionRequest implements IndicesRequest {
+
+        private final String sourceIndex;
+        private final boolean deleteDestIfExists;
```

Comment:
This parameter only exists to test what occurs if there's a race condition and an existing dest index was not deleted (well actually was deleted, but then re-created by another instance of the action). It seems unfortunate to add a parameter for testing, but seemed much simpler than other ways of forcing the race condition. Does this seem okay?